android/java/org/chromium/chrome/browser/toolbar/top/BraveToolbarLayoutImpl.java

```diff
@@ -487,6 +513,12 @@ public void onDidFinishNavigationInPrimaryMainFrame(
                         && mBraveRewardsNativeWorker.IsSupported()) {
                     showBraveRewardsOnboardingModal();
                 }
+                if (ChromeFeatureList.isEnabled(BraveFeatureList.BRAVE_PLAYLIST)
+                        && !tab.getUrl().getSpec().startsWith(UrlConstants.CHROME_SCHEME)
+                        && !UrlUtilities.isNTPUrl(tab.getUrl().getSpec())
+                        && tab.getUrl().domainIs(YOUTUBE_DOMAIN) && mPlaylistService != null) {
+                    // TODO DEEP : find contents from the page and show the playlist button
```

Comment:
FYI, I was thinking that we need PlaylistTabHelper attached to web contentses, which detects files on navigation.

Reply:
Not yet, I just thought about it. If you're under time pressure, please go ahead. We can migrate it later.